smac/utils/validate.py

```diff
@@ -84,35 +81,35 @@ def __init__(self, scenario: Scenario, trajectory: list, output: str,
             num_run = np.random.randint(MAXINT)
             self.rng = np.random.RandomState(seed=num_run)
 
-        self.rh = RunHistory(average_cost)  # update this rh with validation-runs
-
-    def _save_results(self, rh:RunHistory, filename:str):
+    def _save_results(self, rh:RunHistory, output):
         """ Helper to save results to file """
-        if self.output == "":
+        if output == "":
             self.logger.info("No output specified, validated runhistory not saved.")
-        # Check if a folder or a file is specified as output
-        if not self.output.endswith('.json'):
-            old = self.output
-            self.output = os.path.join(self.output, filename)
-            self.logger.debug("Output is \"%s\", changing to \"%s\"!", old,
-                              self.output)
-        base = os.path.split(self.output)[0]
+            return rh
+        base = os.path.split(output)[0]
         if not base == "" and not os.path.exists(base):
             self.logger.debug("Folder (\"%s\") doesn't exist, creating.", base)
             os.makedirs(base)
-        self.logger.info("Saving validation-results in %s", self.output)
-        rh.save_json(self.output)
+        rh.save_json(output)
+        self.logger.info("Saving validation-results in %s", output)
         return rh
 
-    def validate(self, config_mode:Union[str, typing.List[Configuration]]='def',
+    def validate(self,
+                 config_mode:Union[str, typing.List[Configuration]]='def',
                  instance_mode:Union[str, typing.List[str]]='test',
                  repetitions:int=1, n_jobs:int=1, backend:str='threading',
-                 runhistory:RunHistory=None, tae:ExecuteTARun=None):
+                 runhistory:RunHistory=None, tae:ExecuteTARun=None,
+                 output:str="") -> RunHistory:
```

Comment:
For the future, I would suggest to format function arguments differently to make it easier to see the changes:
```python
def validate(
    self,
    arg1,
    arg2,
) -> something:
```
Adding a new argument will only change a single line then.